python/src/scipp/plot/plot_2d.py

```diff
@@ -322,38 +402,53 @@ def slice_data(self):
                     self.slider_label[self.name][dim]["coord"], val.value)
-                self.vslice = self.vslice[val.dim, val.value]
-                # At this point, after masks were combined, all their
-                # dimensions should be contained in the data_array.dims.
-                if self.params["masks"][
-                        self.name]["show"] and dim in self.mslice.dims:
-                    self.mslice = self.mslice[val.dim, val.value]
+                data_slice = data_slice[val.dim, val.value]
+
+                # If a profile viewer is present, we also update the position
+                # of the slice on the profile plot.
+                if self.slice_pos_rectangle is not None:
+                    new_pos = self.slider_coord[self.name][dim][
+                        dim, val.value].value
+                    self.slice_pos_rectangle.set_x(new_pos)
+                    if self.histograms[self.name][dim][dim]:
+                        self.slice_pos_rectangle.set_width(self.slider_coord[
+                            self.name][dim][dim, val.value + 1].value -
+                                                           new_pos)
+                    else:
+                        new_pos -= 0.5 * self.slice_pos_rectangle.get_width()
+                    self.slice_pos_rectangle.set_x(new_pos)
+
         # In the case of unaligned data, we may want to auto-scale the colorbar
         # as we slice through dimensions. Colorbar limits are allowed to grow
         # but not shrink.
-        if self.vslice.unaligned is not None:
-            self.vslice = sc.histogram(self.vslice)
-            self.vslice.variances = None
+        if data_slice.unaligned is not None:
+            data_slice = sc.histogram(data_slice)
+            data_slice.variances = None
             self.autoscale_cbar = True
         else:
             self.autoscale_cbar = False
+
         self.vslice = detail.move_to_data_array(
-            data=sc.Variable(dims=self.vslice.dims,
+            data=sc.Variable(dims=data_slice.dims,
                              unit=sc.units.counts,
-                             values=self.vslice.values,
+                             values=data_slice.values,
+                             variances=data_slice.variances,
                              dtype=sc.dtype.float32))
         self.vslice.coords[self.xyrebin["x"].dims[0]] = self.xyedges["x"]
         self.vslice.coords[self.xyrebin["y"].dims[0]] = self.xyedges["y"]
-        if self.params["masks"][self.name]["show"]:
-            self.vslice.masks["all"] = self.mslice
+
+        # Also include masks
+        if len(data_slice.masks) > 0:
+            for m in data_slice.masks:
+                self.vslice.masks[m] = data_slice.masks[m]
+
         # Scale by bin width and then rebin in both directions
         # Note that this has to be written as 2 inplace operations to avoid
         # creation of large 2D temporary from broadcast
```

Comment:
Do the 2 lines below affect the variances ontop of the `rebin`?

Reply:
No cause variances should be zero for `xywidth['x']` and `xywidth['y']`?

Reply:
Correct, coordinates do not have variances in practice.